app/com/linkedin/drelephant/mapreduce/MapReduceFetcherHadoop2.java

```diff
@@ -328,16 +329,17 @@ private void getTaskDataAll(URL url, String jobId, List<MapReduceTaskData> mappe
 
     private void getTaskData(String jobId, List<MapReduceTaskData> taskList) throws IOException, AuthenticationException {
 
-      // disable sampling for now.
+      int sampleSize = taskList.size();
 
-//      if (taskList.size() > MAX_SAMPLE_SIZE) {
-//        logger.info(jobId + " needs sampling.");
-//        Collections.shuffle(taskList);
-//      }
-//
-//      int sampleSize = Math.min(taskList.size(), MAX_SAMPLE_SIZE);
+      // check if sampling is enabled
+      if(Boolean.parseBoolean(_fetcherConfigurationData.getParamMap().get(SAMPLING_ENABLED))) {
```

Comment:
parseBoolean returns true only if the parameter is not null and ignoring the case is equal to 'true'. In all other cases (including null) it returns false. There won't be any problem because of this.